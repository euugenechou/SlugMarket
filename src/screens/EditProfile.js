/* React imports */
import React from "react";
import {
  TextInput,
  StyleSheet,
  View,
  ScrollView,
  Alert
} from "react-native";
import { API, Auth } from "aws-amplify";

import { Button } from "react-native-elements";

export default class AddItem extends React.Component {
  static navigationOptions = () => ({
    headerTintColor: "teal",
    headerStyle: {
      height: 40,
      backgroundColor: "white",
      shadowColor: "transparent",
      borderBottomWidth: 0
    }
  });

  constructor(props) {
    super(props);
    this.state = { lookingFor: "" };
  }

  onChangeText(key, value) {
    this.setState({
      [key]: value
    });
  }

  // createPost(userId) {
  //   const newPost = { body: this.state };
  //   newPost.body.userId = userId;
  //   newPost.body.timeAdded = new Date().toString();
  //   return newPost;
  // }

  /**
   * Save new post to postedItems schema in DynamoDB
   */
  // async saveItemPost() {
  //   const userInfo = await Auth.currentUserInfo().catch(error =>
  //     Alert.alert(JSON.stringify(error))
  //   );
  //   const path = "/itemPosts";
  //   const postInfo = this.createPost(userInfo["id"]);
  //   try {
  //     const apiResponse = await API.post("itemPostsCRUD", path, postInfo);
  //     Alert.alert(JSON.stringify(apiResponse));
  //   } catch (error) {
  //     Alert.alert(JSON.stringify(error));
  //   }
  // }

  render() {
    return (
      <View style={{ flex: 1 }}>
        <ScrollView
          contentContainerStyle={styles.container}
          centerContent={true}
        >
          <TextInput
            onChangeText={value => this.onChangeText("lookingFor", value)}
            style={styles.input}
            placeholder="What are you looking for?"
            placeholderTextColor="gray"
            autoCapitalize="none"
            autoCorrect={false}
          />
          <Button
            raised
            color="white"
            backgroundColor="teal"
            borderRadius={5}
            containerViewStyle={{ width: 300, paddingTop: 10 }}
            onPress={() => console.log("Changing profile!")}
            title="Save Changes"
            fontWeight="bold"
          />
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  input: {
    height: 50,
    width: 300,
    color: "black",
    borderBottomWidth: 0.8,
    borderBottomColor: "#dddddd",
    marginBottom: 10,
  },
  box: {
    height: 100,
    width: 300,
    borderColor: "gray",
    borderWidth: 3
  },
  container: {
    flex: 1,
    padding: 30,
    backgroundColor: "white",
    alignItems: "center"
  },
  titleText: {
    fontSize: 36,
    padding: 50,
    color: "teal"
  }
});
