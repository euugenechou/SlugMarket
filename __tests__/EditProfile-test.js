// __tests__/EditProfile-test.js
import 'react-native';
import React from 'react';
import EditProfile from '../src/screens/EditProfile';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <EditProfile />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});