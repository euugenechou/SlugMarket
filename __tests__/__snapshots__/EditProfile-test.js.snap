// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    centerContent={true}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "padding": 30,
      }
    }
  >
    <View>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="What are you looking for?"
        placeholderTextColor="gray"
        style={
          Object {
            "borderBottomColor": "#dddddd",
            "borderBottomWidth": 0.8,
            "color": "black",
            "height": 50,
            "marginBottom": 10,
            "width": 300,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "paddingTop": 10,
              "width": 300,
            },
            Object {
              "borderRadius": 5,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "teal",
                },
                Object {
                  "borderRadius": 5,
                },
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "white",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    Object {
                      "fontWeight": "bold",
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              Save Changes
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
